sys/net/gnrc/routing/rpl/of0.c

```diff
@@ -79,11 +81,21 @@ uint16_t calc_rank(gnrc_rpl_parent_t *parent, uint16_t base_rank)
 /* We simply return the Parent with lower rank */
 gnrc_rpl_parent_t *which_parent(gnrc_rpl_parent_t *p1, gnrc_rpl_parent_t *p2)
 {
-    if (p1->rank <= p2->rank) {
-        return p1;
+    if(parent_cmp(p1, p2) == 1) {
```

Comment:
I would recommend to use `if(parent_cmp(p1, p2) > 1)` instead of strict equality here, this leaves more freedom on the implementation of `parent_cmp`. For instance, it might be that `parent_cmp` returns the actual rank difference. Quoting from the `man memcmp`:

> The memcmp() function returns zero if the two strings are identical, otherwise returns the difference between the first two differing bytes (treated as unsigned char values, so that `\200' is greater than `\0', for example).  Zero-length strings are always identical.  This behavior is not required by C and portable code
     should only depend on the sign of the returned value.

the last part is what I'm referring to.